# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: security/manager/ssl/tests/gtest/DeserializeCertTest.cpp
# Commit: 11e5baee978e
# Full Hash: 11e5baee978e8f98fa3795e430dd60c7600c9e8c
# Author: Sean Feng <sefeng@mozilla.com>
# Date: 2019-10-22 09:46:06
# Regressor Bug: 1580315
# File Overlap Count: 1
# Description:
#   Bug 1580315 - Convert certList to raw array for nsITransportSecurityInfo r=keeler,Ehsan,kershaw
#   
#   This patch converts the certList attribute of nsITransportSecurityInfo
#   from nsIX509CertList to Array<nsIx509Cert>
#   
# ==============================================================================

diff -r 341d72ef4bf5 -r 11e5baee978e security/manager/ssl/tests/gtest/DeserializeCertTest.cpp
--- a/security/manager/ssl/tests/gtest/DeserializeCertTest.cpp	Mon Oct 21 19:33:02 2019 +0000
+++ b/security/manager/ssl/tests/gtest/DeserializeCertTest.cpp	Mon Oct 21 19:49:01 2019 +0000
@@ -45,32 +45,18 @@
   ASSERT_EQ(NS_OK, rv);
   ASSERT_TRUE(cert);
 
-  nsCOMPtr<nsIX509CertList> failedChain;
-  rv = securityInfo->GetFailedCertChain(getter_AddRefs(failedChain));
+  nsTArray<RefPtr<nsIX509Cert>> failedCertArray;
+  rv = securityInfo->GetFailedCertChain(failedCertArray);
   ASSERT_EQ(NS_OK, rv);
 
   if (hasFailedCertChain) {
-    ASSERT_TRUE(failedChain);
-    nsCOMPtr<nsISimpleEnumerator> enumerator;
-    rv = failedChain->GetEnumerator(getter_AddRefs(enumerator));
-    ASSERT_EQ(NS_OK, rv);
-    bool hasMore;
-    rv = enumerator->HasMoreElements(&hasMore);
-    ASSERT_EQ(NS_OK, rv);
-    size_t actualFailedCertChainLength = 0;
-    while (hasMore) {
-      nsCOMPtr<nsISupports> supports;
-      rv = enumerator->GetNext(getter_AddRefs(supports));
-      ASSERT_EQ(NS_OK, rv);
-      nsCOMPtr<nsIX509Cert> cert(do_QueryInterface(supports));
-      actualFailedCertChainLength++;
+    ASSERT_FALSE(failedCertArray.IsEmpty());
+    for (const auto& cert : failedCertArray) {
       ASSERT_TRUE(cert);
-      rv = enumerator->HasMoreElements(&hasMore);
-      ASSERT_EQ(NS_OK, rv);
     }
-    ASSERT_EQ(failedCertChainLength, actualFailedCertChainLength);
+    ASSERT_EQ(failedCertChainLength, failedCertArray.Length());
   } else {
-    ASSERT_FALSE(failedChain);
+    ASSERT_TRUE(failedCertArray.IsEmpty());
   }
 }
 